tests/Support/SupportLazyCollectionTest.php

```diff
@@ -69,6 +69,31 @@ public function testCanCreateCollectionFromClosure()
         ], $data->all());
     }
 
+    public function testCanCreateCollectionFromGenerator()
+    {
+        $iterable = function () {
+            yield 1;
+            yield 2;
+            yield 3;
+        };
+        $data = LazyCollection::make($iterable());
+
+        $this->assertSame([1, 2, 3], $data->all());
+
+        $iterable = function () {
+            yield 'a' => 1;
+            yield 'b' => 2;
+            yield 'c' => 3;
+        };
+        $data = LazyCollection::make($iterable());
+
+        $this->assertSame([
+            'a' => 1,
+            'b' => 2,
+            'c' => 3,
+        ], $data->all());
+    }
```

Comment:
BTW, this test passes either way (even without the changes in this PR).

These tests do not check for laziness.